test-apps/imodel-transformer/src/Main.ts

```diff
@@ -122,7 +130,20 @@ void (async () => {
     let targetDb: IModelDb;
     const processChanges = args.sourceStartChangesetIndex || args.sourceStartChangesetId;
 
-    const accessToken = await IModelHubUtils.getAccessToken();
+    let accessToken: AccessToken | undefined;
+    const requiresLogin
+       = args.sourceITwinId
+      || args.targetITwinId
+      || args.sourceIModelId
+      || args.sourceIModelName
+      || args.sourceStartChangesetId
+      || args.sourceStartChangesetIndex
+      || args.targetIModelId
+      || args.targetIModelName;
+
+    if (requiresLogin)
+      accessToken = await IModelHubUtils.getAccessToken();
```

Comment:
```ts
const accessToken = requiresLoging ? await  IModelHubUtils.getAccessToken() : "";
```

Reply:
I am currently in process of rewriting it to not need any assertions at all, by lazily async fetching the accessToken. I will push the update soon

Reply:
when not logged in, AccessToken should be "";

Reply:
this should now be the case because getAccessToken is used directly everywhere

Reply:
I missed that you were calling `IModelHubUtils.getAccessToken` (a function which should not exist), not `IModelHost.getAccessToken`. 

Somewhere, you need to call and await `IModelHubUtils.signIn` if you require login. Maybe that's already happening?

Reply:
I missed it too. `IModelHubUtils.getAccessToken` has a few problems, and I am going to remove it and replace it with direct usage.

Reply:
I removed the custom getAccessToken implementation in IModelHubUtils but chose to continue using a custom local getAccessToken function, now it's `IModelTransformerTestAppHost.getAccessToken`(this app's local way to call `IModelHost.authorizationBackend.getAccessToken()`) which will log in when called for the first time, and call `getAccessToken` directly otherwise.

If I didn't sign in lazily, I'd either need to have nullable state like I did before, which means I would have to cast everything because typescript can't infer that the accessToken is non null when certain other options are set,
or I'd need to separate the offline and the online logic in Main.ts, which imo would make it less readable by coupling auth to the preparing of the source and target iModels.

Please comment/resolve.